backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/dialect/model/impl/OpenSearch2ModelDialect.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.backend.elasticsearch.dialect.model.impl;
+
+import org.hibernate.search.backend.elasticsearch.types.dsl.provider.impl.ElasticsearchIndexFieldTypeFactoryProvider;
+import org.hibernate.search.backend.elasticsearch.types.dsl.provider.impl.OpenSearch1IndexFieldTypeFactoryProvider;
+
+import com.google.gson.Gson;
+
+/**
+ * The model dialect for OpenSearch 2.x.
+ */
+public class OpenSearch2ModelDialect implements ElasticsearchModelDialect {
```

Comment:
I'm not sure that we need `OpenSearch1` and `OpenSearch2` as they are identical, I've just followed the Elasticsearch example.

Reply:
I highly doubt that OpenSearch 1.x supports vector search, so you'll probably need to make them different, be it only to throw a clear error when people try to use vector search on older versions of OpenSearch/Elasticsearch.

Reply:
I hoped that it would work since there's a doc for it https://opensearch.org/docs/1.3/search-plugins/knn/knn-index in 1.3 version... But I've run a quick test and it results in:
```json
{
  "error": {
    "root_cause": [
      {
        "type": "mapper_parsing_exception",
        "reason": "Invalid engine: lucene"
      }
    ],
    "type": "mapper_parsing_exception",
    "reason": "Failed to parse mapping [_doc]: Invalid engine: lucene",
    "caused_by": {
      "type": "mapper_parsing_exception",
      "reason": "Invalid engine: lucene"
    }
  },
  "status": 400
}
```
even though there's an example in that doc 🤷🏻‍♂️😔:
```
{
    "type": "knn_vector",
    "dimension": 100,
    "method": {
        "name":"hnsw",
        "engine":"lucene",
        "space_type": "l2",
        "parameters":{
            "m":2048,
            "ef_construction": 245
        }
    }
}
```
Then, on the Elasticsearch side, from what I see, this API has been available as a preview since 8.0

Reply:
IMO you shouldn't bother too much with older releases. If it takes longer than just implementing clear "Feature not available on this version" error messages, your time is better invested elsewhere :) It is an incubating feature, after all.